File: MySmartNews/objective-c/ChildViewController.m
//
//  ViewController.m
//  TabbedPageViewController
//
//  Created by Merrick Sapsford on 24/12/2015.
//  Copyright © 2015 Merrick Sapsford. All rights reserved.
//

#import "ChildViewController.h"
#import "DDXML.h"
#import "NewsViewController.h"

@implementation ChildViewController
{
    
    NSMutableArray *titleArray;
    NSMutableArray *linkArray;
    NSMutableArray *pubDateArray;
    int pagenumber;


    NSString *urlString;

}

- (void)viewDidLoad {
    [super viewDidLoad];
//    self.titleLabel.text = [NSString stringWithFormat:@"Page %i", (int)(self.pageIndex + 1)];
    
    
    self.table.dataSource = self;
    self.table.delegate = self;
    
    [self.table reloadData];

}

-(void)viewWillAppear:(BOOL)animated{

    [super viewWillAppear:animated];
    
   pagenumber = (int)self.pageIndex + 1;
    
    if (pagenumber == 1) {
        
        
        UIImage *image = [UIImage imageNamed:@"bg1.jpg"];
        UIImageView *backGroungImageView = [[UIImageView alloc]initWithImage:image];
        self.table.backgroundView = backGroungImageView;
        
        [self getXML1];
    }
    if (pagenumber == 2) {
        
        
        UIImage *image = [UIImage imageNamed:@"bg2.jpg"];
        UIImageView *backGroungImageView = [[UIImageView alloc]initWithImage:image];
        self.table.backgroundView = backGroungImageView;
        
        [self getXML2];
    }
    if (pagenumber == 3) {
        
        
        UIImage *image = [UIImage imageNamed:@"bg3.jpg"];
        UIImageView *backGroungImageView = [[UIImageView alloc]initWithImage:image];
        self.table.backgroundView = backGroungImageView;
        
        [self getXML3];
    }
    if (pagenumber == 4) {
        
        
        UIImage *image = [UIImage imageNamed:@"bg4.jpg"];
        UIImageView *backGroungImageView = [[UIImageView alloc]initWithImage:image];
        self.table.backgroundView = backGroungImageView;
        
        [self getXML4];
    }
    
    if (pagenumber == 5) {
        
        
        UIImage *image = [UIImage imageNamed:@"bg5.jpg"];
        UIImageView *backGroungImageView = [[UIImageView alloc]initWithImage:image];
        self.table.backgroundView = backGroungImageView;
        
        [self getXML5];
    }
    if (pagenumber == 6) {
        
        
        UIImage *image = [UIImage imageNamed:@"bg6.jpg"];
        UIImageView *backGroungImageView = [[UIImageView alloc]initWithImage:image];
        self.table.backgroundView = backGroungImageView;
        
        [self getXML6];
    }
    
    [self.table reloadData];
    
    
}

//セクション数を決めるメソッド
- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    
    return 1;
}


//セクションの中にあるセルの数の設定
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    
    //配列の中身の数
    return titleArray.count;
}

//セルを作成するデリゲート
- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    
    static NSString *CellIdentifier = @"Cell";
    // tableCell の ID で UITableViewCell のインスタンスを生成
    UITableViewCell *cell =
    [tableView dequeueReusableCellWithIdentifier:CellIdentifier];
    
    if(cell==nil){
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleSubtitle reuseIdentifier:CellIdentifier];
    }
    
    // Configure the cell...
    
    //タグでひも付けする
    UILabel *titleLabel = (UILabel *)[cell viewWithTag:1];
    UILabel *dateLabel = (UILabel *)[cell viewWithTag:2];
    
    titleLabel.text = [titleArray objectAtIndex:indexPath.row];
    dateLabel.text = [pubDateArray objectAtIndex:indexPath.row];
    
    
    [cell.contentView addSubview:titleLabel];
    [cell.contentView addSubview:dateLabel];
    
    
    return cell;
}





-(void)getXML1{

    //ここでRSS情報を取得する
    
    titleArray = [NSMutableArray array];
    linkArray = [NSMutableArray array];
    pubDateArray = [NSMutableArray array];
    
    //エラー処理をするための変数を準備する(一応)
    
    NSError *error = nil;
    NSURL *url = [NSURL URLWithString:@"http://rss.dailynews.yahoo.co.jp/fc/rss.xml"];
    
    NSData *data = [[NSMutableData alloc]initWithContentsOfURL:url];
    

    DDXMLDocument *doc = [[DDXMLDocument alloc]initWithData:data options:0 error:&error];
    
    NSArray *nodes = [NSArray array];
    nodes = [doc nodesForXPath:@"//item" error:nil];

    
    for(DDXMLElement *node in nodes){
    
        NSString *title = [[node childAtIndex:0] stringValue];
        NSString *link = [[node childAtIndex:1] stringValue];
        NSString *pub = [[node childAtIndex:2] stringValue];
        
        [titleArray addObject:title];
        [linkArray addObject:link];
        [pubDateArray addObject:pub];
        
        
    }
    
    
}





-(void)getXML2{
    
    //ここでRSS情報を取得する
    
    titleArray = [NSMutableArray array];
    linkArray = [NSMutableArray array];
    pubDateArray = [NSMutableArray array];
    
    //エラー処理をするための変数を準備する(一応)
    
    NSError *error = nil;
    NSURL *url = [NSURL URLWithString:@"http://headlines.yahoo.co.jp/rss/bfj-c_ent.xml"];
    
    NSData *data = [[NSMutableData alloc]initWithContentsOfURL:url];
    
    
    DDXMLDocument *doc = [[DDXMLDocument alloc]initWithData:data options:0 error:&error];
    
    NSArray *nodes = [NSArray array];
    nodes = [doc nodesForXPath:@"//item" error:nil];
    
    
    for(DDXMLElement *node in nodes){
        
        NSString *title = [[node childAtIndex:0] stringValue];
        NSString *link = [[node childAtIndex:1] stringValue];
        NSString *pub = [[node childAtIndex:3] stringValue];
        
        [titleArray addObject:title];
        [linkArray addObject:link];
        [pubDateArray addObject:pub];
        
        
    }

    
}

-(void)getXML3{
    
    //ここでRSS情報を取得する
    
    titleArray = [NSMutableArray array];
    linkArray = [NSMutableArray array];
    pubDateArray = [NSMutableArray array];
    
    //エラー処理をするための変数を準備する(一応)
    
    NSError *error = nil;
    NSURL *url = [NSURL URLWithString:@"http://headlines.yahoo.co.jp/rss/zdn_m-c_ent.xml"];
    
    NSData *data = [[NSMutableData alloc]initWithContentsOfURL:url];
    
    
    DDXMLDocument *doc = [[DDXMLDocument alloc]initWithData:data options:0 error:&error];
    
    NSArray *nodes = [NSArray array];
    nodes = [doc nodesForXPath:@"//item" error:nil];
    
    
    for(DDXMLElement *node in nodes){
        
        NSString *title = [[node childAtIndex:0] stringValue];
        NSString *link = [[node childAtIndex:1] stringValue];
        NSString *pub = [[node childAtIndex:3] stringValue];
        
        [titleArray addObject:title];
        [linkArray addObject:link];
        [pubDateArray addObject:pub];
        
        
    }

    
}
-(void)getXML4{
    
    //ここでRSS情報を取得する
    
    titleArray = [NSMutableArray array];
    linkArray = [NSMutableArray array];
    pubDateArray = [NSMutableArray array];
    
    //エラー処理をするための変数を準備する(一応)
    
    NSError *error = nil;
    NSURL *url = [NSURL URLWithString:@"http://headlines.yahoo.co.jp/rss/tospoweb-c_ent.xml"];
    
    NSData *data = [[NSMutableData alloc]initWithContentsOfURL:url];
    
    
    DDXMLDocument *doc = [[DDXMLDocument alloc]initWithData:data options:0 error:&error];
    
    NSArray *nodes = [NSArray array];
    nodes = [doc nodesForXPath:@"//item" error:nil];
    
    
    for(DDXMLElement *node in nodes){
        
        NSString *title = [[node childAtIndex:0] stringValue];
        NSString *link = [[node childAtIndex:1] stringValue];
        NSString *pub = [[node childAtIndex:3] stringValue];
        
        [titleArray addObject:title];
        [linkArray addObject:link];
        [pubDateArray addObject:pub];
        
        
    }
    

    
    
}
-(void)getXML5{
    
    //ここでRSS情報を取得する
    
    titleArray = [NSMutableArray array];
    linkArray = [NSMutableArray array];
    pubDateArray = [NSMutableArray array];
    
    //エラー処理をするための変数を準備する(一応)
    
    NSError *error = nil;
    NSURL *url = [NSURL URLWithString:@"http://headlines.yahoo.co.jp/rss/it_nlab-c_ent.xml"];
    
    NSData *data = [[NSMutableData alloc]initWithContentsOfURL:url];
    
    
    DDXMLDocument *doc = [[DDXMLDocument alloc]initWithData:data options:0 error:&error];
    
    NSArray *nodes = [NSArray array];
    nodes = [doc nodesForXPath:@"//item" error:nil];
    
    
    for(DDXMLElement *node in nodes){
        
        NSString *title = [[node childAtIndex:0] stringValue];
        NSString *link = [[node childAtIndex:1] stringValue];
        NSString *pub = [[node childAtIndex:3] stringValue];
        
        [titleArray addObject:title];
        [linkArray addObject:link];
        [pubDateArray addObject:pub];
        
        
    }
    
}
-(void)getXML6{
    
    //ここでRSS情報を取得する
    
    titleArray = [NSMutableArray array];
    linkArray = [NSMutableArray array];
    pubDateArray = [NSMutableArray array];
    
    //エラー処理をするための変数を準備する(一応)
    
    NSError *error = nil;
    NSURL *url = [NSURL URLWithString:@"http://headlines.yahoo.co.jp/rss/zdn_mkt-dom.xml"];
    
    NSData *data = [[NSMutableData alloc]initWithContentsOfURL:url];
    
    
    DDXMLDocument *doc = [[DDXMLDocument alloc]initWithData:data options:0 error:&error];
    
    NSArray *nodes = [NSArray array];
    nodes = [doc nodesForXPath:@"//item" error:nil];
    
    
    for(DDXMLElement *node in nodes){
        
        NSString *title = [[node childAtIndex:0] stringValue];
        NSString *link = [[node childAtIndex:1] stringValue];
        NSString *pub = [[node childAtIndex:3] stringValue];
        
        [titleArray addObject:title];
        [linkArray addObject:link];
        [pubDateArray addObject:pub];
        
        
    }

    
}





-(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{

    
    urlString = [linkArray objectAtIndex:indexPath.row];
    [self performSegueWithIdentifier:@"webNews" sender:self];

}



-(void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender{


    if ([[segue identifier] isEqualToString:@"webNews"]) {

        NewsViewController *newsViewController = (NewsViewController *)[segue destinationViewController];
        newsViewController.urlString = urlString;
    }

}












@end

File: MySmartNews/ChildViewController.swift
//
//  ChildViewController.swift
//  Paged Tabs Example
//
//  Created by Merrick Sapsford on 04/05/2016.
//  Copyright © 2016 Merrick Sapsford. All rights reserved.
//

import Foundation
import UIKit
import DDXML.h
import NewsViewController.h

class ChildViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
    @IBOutlet weak var table: UITableView!
    
    private var titleArray = [AnyHashable]()
    private var linkArray = [AnyHashable]()
    private var pubDateArray = [AnyHashable]()
    private var pagenumber: Int = 0
    private var urlString = ""
    
    func viewDidLoad() {
        super.viewDidLoad()
        //    self.titleLabel.text = [NSString stringWithFormat:@"Page %i", (int)(self.pageIndex + 1)];
        table.dataSource = self
        table.delegate = self
        table.reloadData()
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        pagenumber = Int(pageIndex) + 1
        if pagenumber == 1 {
            let image = UIImage(named: "bg1.jpg")
            let backGroungImageView = UIImageView(image: image)
            table.backgroundView = backGroungImageView
            getXML1()
        }
        if pagenumber == 2 {
            let image = UIImage(named: "bg2.jpg")
            let backGroungImageView = UIImageView(image: image)
            table.backgroundView = backGroungImageView
            getXML2()
        }
        if pagenumber == 3 {
            let image = UIImage(named: "bg3.jpg")
            let backGroungImageView = UIImageView(image: image)
            table.backgroundView = backGroungImageView
            getXML3()
        }
        if pagenumber == 4 {
            let image = UIImage(named: "bg4.jpg")
            let backGroungImageView = UIImageView(image: image)
            table.backgroundView = backGroungImageView
            getXML4()
        }
        if pagenumber == 5 {
            let image = UIImage(named: "bg5.jpg")
            let backGroungImageView = UIImageView(image: image)
            table.backgroundView = backGroungImageView
            getXML5()
        }
        if pagenumber == 6 {
            let image = UIImage(named: "bg6.jpg")
            let backGroungImageView = UIImageView(image: image)
            table.backgroundView = backGroungImageView
            getXML6()
        }
        table.reloadData()
    }
    
    //セクション数を決めるメソッド
    func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    //セクションの中にあるセルの数の設定
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        //配列の中身の数
        return titleArray.count
    }
    
    //セルを作成するデリゲート
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let CellIdentifier = "Cell"
        // tableCell の ID で UITableViewCell のインスタンスを生成
        var cell: UITableViewCell? = tableView.dequeueReusableCell(withIdentifier: CellIdentifier)
        if cell == nil {
            cell = UITableViewCell(style: .subtitle, reuseIdentifier: CellIdentifier)
        }
        // Configure the cell...
        //タグでひも付けする
        let titleLabel = cell?.viewWithTag(1) as? UILabel
        let dateLabel = cell?.viewWithTag(2) as? UILabel
        titleLabel?.text = titleArray[indexPath.row] as? String
        dateLabel?.text = pubDateArray[indexPath.row] as? String
        cell?.contentView.addSubview(titleLabel ?? UIView())
        cell?.contentView.addSubview(dateLabel ?? UIView())
        return cell ?? UITableViewCell()
    }
    
    func getXML1() {
        //ここでRSS情報を取得する
        titleArray = [AnyHashable]()
        linkArray = [AnyHashable]()
        pubDateArray = [AnyHashable]()
        //エラー処理をするための変数を準備する(一応)
        var error: Error? = nil
        let url = URL(string: "http://rss.dailynews.yahoo.co.jp/fc/rss.xml")
        let data = Data(contentsOf: url!)
        let doc = try? DDXMLDocument(data: data, options: 0)
        var nodes = [Any]()
        nodes = (try? doc?.nodes(forXPath: "//item")) ?? [Any]()
        for node: DDXMLElement in nodes {
            let title = "\(node.child(at: 0))"
            let link = "\(node.child(at: 1))"
            let pub = "\(node.child(at: 2))"
            titleArray.append(title)
            linkArray.append(link)
            pubDateArray.append(pub)
        }
    }
    
    func getXML2() {
        //ここでRSS情報を取得する
        titleArray = [AnyHashable]()
        linkArray = [AnyHashable]()
        pubDateArray = [AnyHashable]()
        //エラー処理をするための変数を準備する(一応)
        var error: Error? = nil
        let url = URL(string: "http://headlines.yahoo.co.jp/rss/bfj-c_ent.xml")
        let data = Data(contentsOf: url!)
        let doc = try? DDXMLDocument(data: data, options: 0)
        var nodes = [Any]()
        nodes = (try? doc?.nodes(forXPath: "//item")) ?? [Any]()
        for node: DDXMLElement in nodes {
            let title = "\(node.child(at: 0))"
            let link = "\(node.child(at: 1))"
            let pub = "\(node.child(at: 3))"
            titleArray.append(title)
            linkArray.append(link)
            pubDateArray.append(pub)
        }
    }
    
    func getXML3() {
        //ここでRSS情報を取得する
        titleArray = [AnyHashable]()
        linkArray = [AnyHashable]()
        pubDateArray = [AnyHashable]()
        //エラー処理をするための変数を準備する(一応)
        var error: Error? = nil
        let url = URL(string: "http://headlines.yahoo.co.jp/rss/zdn_m-c_ent.xml")
        let data = Data(contentsOf: url!)
        let doc = try? DDXMLDocument(data: data, options: 0)
        var nodes = [Any]()
        nodes = (try? doc?.nodes(forXPath: "//item")) ?? [Any]()
        for node: DDXMLElement in nodes {
            let title = "\(node.child(at: 0))"
            let link = "\(node.child(at: 1))"
            let pub = "\(node.child(at: 3))"
            titleArray.append(title)
            linkArray.append(link)
            pubDateArray.append(pub)
        }
    }
    
    func getXML4() {
        //ここでRSS情報を取得する
        titleArray = [AnyHashable]()
        linkArray = [AnyHashable]()
        pubDateArray = [AnyHashable]()
        //エラー処理をするための変数を準備する(一応)
        var error: Error? = nil
        let url = URL(string: "http://headlines.yahoo.co.jp/rss/tospoweb-c_ent.xml")
        let data = Data(contentsOf: url!)
        let doc = try? DDXMLDocument(data: data, options: 0)
        var nodes = [Any]()
        nodes = (try? doc?.nodes(forXPath: "//item")) ?? [Any]()
        for node: DDXMLElement in nodes {
            let title = "\(node.child(at: 0))"
            let link = "\(node.child(at: 1))"
            let pub = "\(node.child(at: 3))"
            titleArray.append(title)
            linkArray.append(link)
            pubDateArray.append(pub)
        }
    }
    
    func getXML5() {
        //ここでRSS情報を取得する
        titleArray = [AnyHashable]()
        linkArray = [AnyHashable]()
        pubDateArray = [AnyHashable]()
        //エラー処理をするための変数を準備する(一応)
        var error: Error? = nil
        let url = URL(string: "http://headlines.yahoo.co.jp/rss/it_nlab-c_ent.xml")
        let data = Data(contentsOf: url!)
        let doc = try? DDXMLDocument(data: data, options: 0)
        var nodes = [Any]()
        nodes = (try? doc?.nodes(forXPath: "//item")) ?? [Any]()
        for node: DDXMLElement in nodes {
            let title = "\(node.child(at: 0))"
            let link = "\(node.child(at: 1))"
            let pub = "\(node.child(at: 3))"
            titleArray.append(title)
            linkArray.append(link)
            pubDateArray.append(pub)
        }
    }
    
    

    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        urlString = linkArray[indexPath.row] as? String ?? ""
        performSegue(withIdentifier: "webNews", sender: self)
    }
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if (segue.identifier == "webNews") {
            let newsViewController = segue.destination as? NewsViewController
            newsViewController?.urlString = urlString
        }
    }
}

